src/daemon.js

```diff
@@ -7,101 +7,49 @@ const multiaddr = require('multiaddr')
 
-const exec = require('./exec')
-
-const ipfsDefaultPath = findIpfsExecutable()
-
-const GRACE_PERIOD = 7500 // amount of ms to wait before sigkill
-
-function findIpfsExecutable () {
-  const rootPath = process.env.testpath ? process.env.testpath : __dirname
-
-  let appRoot = path.join(rootPath, '..')
-  // If inside <appname>.asar try to load from .asar.unpacked
-  // this only works if asar was built with
-  // asar --unpack-dir=node_modules/go-ipfs-dep/* (not tested)
-  // or
-  // electron-packager ./ --asar.unpackDir=node_modules/go-ipfs-dep
-  if (appRoot.includes(`.asar${path.sep}`)) {
-    appRoot = appRoot.replace(`.asar${path.sep}`, `.asar.unpacked${path.sep}`)
-  }
-  const appName = isWindows ? 'ipfs.exe' : 'ipfs'
-  const depPath = path.join('go-ipfs-dep', 'go-ipfs', appName)
-  const npm3Path = path.join(appRoot, '../', depPath)
-  const npm2Path = path.join(appRoot, 'node_modules', depPath)
+const tryJsonParse = utils.tryJsonParse
+const parseConfig = utils.parseConfig
+const tempDir = utils.tempDir
+const findIpfsExecutable = utils.findIpfsExecutable
+const setConfigValue = utils.setConfigValue
+const configureNode = utils.configureNode
 
-  if (fs.existsSync(npm3Path)) {
-    return npm3Path
-  }
-  if (fs.existsSync(npm2Path)) {
-    return npm2Path
-  }
-
-  throw new Error('Cannot find the IPFS executable')
-}
-
-function setConfigValue (node, key, value, callback) {
-  exec(
-    node.exec,
-    ['config', key, value, '--json'],
-    { env: node.env },
-    callback
-  )
-}
-
-function configureNode (node, conf, callback) {
-  async.eachOfSeries(conf, (value, key, cb) => {
-    setConfigValue(node, key, JSON.stringify(value), cb)
-  }, callback)
-}
+const exec = require('./exec')
 
-function tryJsonParse (input, callback) {
-  let res
-  try {
-    res = JSON.parse(input)
-  } catch (err) {
-    return callback(err)
-  }
-  callback(null, res)
-}
-
-// Consistent error handling
-function parseConfig (path, callback) {
-  async.waterfall([
-    (cb) => fs.readFile(join(path, 'config'), cb),
-    (file, cb) => tryJsonParse(file.toString(), cb)
-  ], callback)
-}
+const GRACE_PERIOD = 10500 // amount of ms to wait before sigkill
 
 /**
- * Controll a go-ipfs node.
+ * Controll a go-ipfs or js-ipfs node.
  */
 class Node {
   /**
    * Create a new node.
    *
-   * @param {string} path
    * @param {Object} [opts]
    * @param {Object} [opts.env={}] - Additional environment settings, passed to executing shell.
-   * @param {boolean} [disposable=false] - Should this be a temporary node.
    * @returns {Node}
    */
-  constructor (path, opts, disposable) {
-    this.path = path
-    this.opts = opts || {}
-    this.exec = process.env.IPFS_EXEC || ipfsDefaultPath
+  constructor (opts) {
+    const rootPath = process.env.testpath ? process.env.testpath : __dirname
+    const isJs = truthy(process.env.IPFS_JS)
+
+    this.opts = opts || { isJs: isJs || false }
+    process.env.IPFS_JS = this.opts.isJs
```

Comment:
Use env vars as read only. This will mess the spawn of multiple nodes in one go.